calicoctl/commands/node/diags.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package node
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docopt/docopt-go"
+)
+
+// diagCmds is a slice of struct to hold commands, their info and filename to run diagnostic on
+type diagCmds []struct {
```

Comment:
Wouldn't this be better as just a single command rather than a []?  That way you can simply pass the structure straight through to the `writeDiags` function rather than pulling out each field from the anonymous inner structure.